interp/interp.go

```diff
@@ -314,6 +321,28 @@ func (interp *Interpreter) Eval(src string) (reflect.Value, error) {
 	return res, err
 }
 
+// Stop stops the execution of the interpreter
+func (interp *Interpreter) Stop() { interp.runid++ }
+
+// EvalTimeout is the same as Eval, except it will return after a timeout it the
+// execution is not complete.
+func (interp *Interpreter) EvalTimeout(src string, d time.Duration) (v reflect.Value, err error) {
+	done := make(chan struct{})
+
+	go func() {
+		defer close(done)
+		v, err = interp.Eval(src)
+	}()
+
+	select {
+	case <-time.After(d):
```

Comment:
Make this a `ctx.Done()` call, then cancellation, timeouts and the safer time cancellation approach that is deadlines.